### ISSUE TITLE
Ability to deselect item

### ISSUE THREAD

**Issue by paulschreiber** [pstanoev/simple-svelte-autocomplete#17]
Once you select an item, there is no way to unselect it. It would be handy to include a way to clear the selection (perhaps an "X").

![image](https://user-images.githubusercontent.com/86784/95618842-b3c03b00-0a3b-11eb-85f5-d3b9c55411d9.png)

**Comment by paulschreiber** [comment]
Here's roughly what I did:

```diff
   .autocomplete-input {
     font: inherit;
-    width: 100%;
+    width: calc(100% - 2rem);
     height: 100%;
     padding: 5px 11px;
   }
```

```diff
  .select:not(.is-multiple):not(.is-loading)::after {
-    right: 1.125em;
+    right: 3.125em;
```

```diff
+function clearValue() {
+    text = '';
+}
```

```diff
+  .remove {
+    font-family: sans-serif; /* override bootstrap */
+    color: white;
+    background-color: grey;
+    cursor: pointer;
+    border-radius: 30px;
+    height: 1.3em;
+    width: 1.3em;
+    display: inline-block;
+    text-align: center;
+    line-height: 22px;
+  }
-<div class="{className} autocomplete select is-fullwidth {uniqueId}">
-  <input
-    type="text"
-    class="input autocomplete-input"
-    {placeholder}
-    {name}
-    {disabled}
-    {title}
-    bind:this={input}
-    bind:value={text}
-    on:input={onInput}
-    on:focus={onFocus}
-    on:keydown={onKeyDown}
-    on:click={onInputClick}
-    on:keypress={onKeyPress} />
+<div class="{className} autocomplete select is-fullwidth {uniqueId}">
+  <span class="remove-wrapper">
+    <input
+      type="text"
+      class="input autocomplete-input"
+      {placeholder}
+      {name}
+      {disabled}
+      {title}
+      bind:this={input}
+      bind:value={text}
+      on:input={onInput}
+      on:focus={onFocus}
+      on:keydown={onKeyDown}
+      on:click={onInputClick}
+      on:keypress={onKeyPress} />
+      <span on:click={clearValue} class="remove">&#10006;</span>
+  </span>
```

**Comment by pstanoev** [comment]
Yes, that feature is missing. I will think about it and likely include something.

**Comment by paulschreiber** [comment]
Let me know if you want a PR, or if the above diff is ok. My fork has deviated a bit from yours at the moment.

**Comment by pstanoev** [comment]
The diff is OK, thanks.

**Comment by pstanoev** [comment]
Implemented in 1.2.4.
